src/DIRAC/RequestManagementSystem/private/RequestTask.py

```diff
@@ -290,11 +286,15 @@ def __call__(self):
           for opFile in operation:
             opFile.Status = 'Failed'
           operation.Status = 'Failed'
+      elif userSuspended in setupProxy['Message']:
```

Comment:
Ouf... can't we use a proper `DErrno` here?

Reply:
No idea!

Reply:
It comes from far away in the Core of DIRAC... I am afraid to pull an infinite string ;-)

Reply:
If you don't mind, I can just push a fix on your branch, and then you can use `cmpError(setupProxy, EVOMS)`

```python
In [16]: res = gProxyManager.downloadVOMSProxyToFile(ownerDN, ownerGroup)

In [17]: print(res)
{'CallStack': ['  File "/opt/dirac/pro/diracos/usr/lib64/python2.7/threading.py", line 777, in __bootstrap\n    self.__bootstrap_inner()\n', '  File "/opt/dirac/pro/diracos/usr/lib64/python2.7/threading.py", line 804, in __bootstrap_inner\n    self.run()\n', '  File "/opt/dirac/pro/diracos/usr/lib64/python2.7/threading.py", line 757, in run\n    self.__target(*self.__args, **self.__kwargs)\n', '  File "/opt/dirac/pro/diracos/usr/lib64/python2.7/site-packages/concurrent/futures/thread.py", line 75, in _worker\n    work_item.run()\n', '  File "/opt/dirac/pro/diracos/usr/lib64/python2.7/site-packages/concurrent/futures/thread.py", line 63, in run\n    result = self.fn(*self.args, **self.kwargs)\n', '  File "/opt/dirac/pro/DIRAC/Core/DISET/private/Service.py", line 436, in _processInThread\n    result = self._processProposal(trid, proposalTuple, handlerObj)\n', '  File "/opt/dirac/pro/DIRAC/Core/DISET/private/Service.py", line 610, in _processProposal\n    result = self._executeAction(trid, proposalTuple, handlerObj)\n', '  File "/opt/dirac/pro/DIRAC/Core/DISET/private/Service.py", line 630, in _executeAction\n    response = handlerObj._rh_executeAction(proposalTuple)\n', '  File "/opt/dirac/pro/DIRAC/Core/DISET/RequestHandler.py", line 129, in _rh_executeAction\n    retVal = self.__doRPC(actionTuple[1])\n', '  File "/opt/dirac/pro/DIRAC/Core/DISET/RequestHandler.py", line 250, in __doRPC\n    return self.__RPCCallFunction(method, args)\n', '  File "/opt/dirac/pro/DIRAC/Core/DISET/RequestHandler.py", line 291, in __RPCCallFunction\n    uReturnValue = oMethod(*args)\n', '  File "/opt/dirac/pro/DIRAC/FrameworkSystem/Service/ProxyManagerHandler.py", line 244, in export_getVOMSProxy\n    return self.__getVOMSProxy(userDN, userGroup, requestPem, requiredLifetime, vomsAttribute, forceLimited)\n', '  File "/opt/dirac/pro/DIRAC/FrameworkSystem/Service/ProxyManagerHandler.py", line 249, in __getVOMSProxy\n    requestedVOMSAttr=vomsAttribute)\n', '  File "/opt/dirac/pro/DIRAC/FrameworkSystem/DB/ProxyDB.py", line 896, in getVOMSProxy\n    return S_ERROR(DErrno.EVOMS, "Cannot append voms extension: %s" % retVal[\'Message\'])\n'], 'Errno': 1121, 'Message': 'VOMS Error ( 1121 : Cannot append voms extension: VOMS Error ( 1121 : Failed to set VOMS attributes. Command: voms-proxy-init2 -cert "/tmp/tmpsoH3SH" -key "/tmp/tmpsoH3SH" -out "/tmp/tmpfzFbNr" -voms "lhcb:/lhcb/Role=user" -valid "688:19" -vomses "/etc/vomses" -r -timeout 12; StdOut: Your identity: /DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=wangya/CN=843467/CN=Yang Wang/CN=8998912830\nCreating temporary proxy  Done\nContacting  voms2.cern.ch:15003 [/DC=ch/DC=cern/OU=computers/CN=voms2.cern.ch] "lhcb" Failed\n\nTrying next server for lhcb.\nCreating temporary proxy  Done\nContacting  lcg-voms2.cern.ch:15003 [/DC=ch/DC=cern/OU=computers/CN=lcg-voms2.cern.ch] "lhcb" Failed\n; StdErr: .....................................................................................................................................\nError: User is currently suspended!\nSuspension reason: User failed to sign the AUP in time.\n................................\nError: User is currently suspended!\nSuspension reason: User failed to sign the AUP in time.\n\nNone of the contacted servers for lhcb were capable\nof returning a valid AC for the user.\n))', 'OK': False, 'rpcStub': [['Framework/ProxyManager', {'timeout': 120, 'skipCACheck': False, 'keepAliveLapse': 150}], 'getVOMSProxy', ['/DC=ch/DC=cern/OU=Organic Units/OU=Users/CN=wangya/CN=843467/CN=Yang Wang', 'lhcb_user', b'-----BEGIN CERTIFICATE REQUEST-----\nMIIBTzCBuQIBADAQMQ4wDAYDVQQDDAVwcm94eTCBnzANBgkqhkiG9w0BAQEFAAOB\njQAwgYkCgYEAwErT5bMlSpPo3tHfUGSO22TAPWKn9ksWzf9RxvPbzy/F4gn8Yu7g\n/rf9fnwhYdz2LJz+spnHwLXr2fpC7py+qf0xuhYSXGrDdD4hIpJvNsAS7ez1yTfx\nlJuT634ufIISbNEEVnPpzFPK9ossMKKFC1FI+9vGXyBzyEnCgX7qyXMCAwEAAaAA\nMA0GCSqGSIb3DQEBCwUAA4GBAD5lwKkmU+I6g/D7mH5P8McUAaADC9jAf/Ef7845\n6tSikx78s/+24B+o6eiUR06H9/cI0T0qMlqkjtLAnITVF7rMGsdDf7X4W6JFqxjo\nPFniRiMap/MWcVnXcP4pCZhIADVe396yt0gvRWK+5pfRCUPF0gtmYzpGYbCH4q1j\nElrQ\n-----END CERTIFICATE REQUEST-----\n', 15600, None]]}

In [18]: cmpError(res, EVOMS)
Out[18]: True
```

Reply:
Once https://github.com/DIRACGrid/DIRAC/pull/5440 is  merge, you could rely on this and use `EVOMS` to compare. 
You can also add

```python
--- a/src/DIRAC/RequestManagementSystem/private/RequestTask.py
+++ b/src/DIRAC/RequestManagementSystem/private/RequestTask.py
@@ -32,6 +32,7 @@ import time
 
 # # from DIRAC
 from DIRAC import gLogger, S_OK, S_ERROR, gConfig
+from DIRAC.Core.Utilities import DErrno
 from DIRAC.FrameworkSystem.Client.MonitoringClient import gMonitor
 from DIRAC.RequestManagementSystem.Client.ReqClient import ReqClient
 from DIRAC.RequestManagementSystem.Client.Request import Request
@@ -283,7 +284,8 @@ class RequestTask(object):
     setupProxy = self.setupProxy()
     if not setupProxy["OK"]:
       self.request.Error = setupProxy["Message"]
-      if 'has no proxy registered' in setupProxy["Message"]:
+      # In case the user does not have proxy
+      if DErrno.cmpError(setupProxy, DErrno.EPROXYFIND):
         self.log.error('Error setting proxy. Request set to Failed:', setupProxy["Message"])
         # If user is no longer registered, fail the request
         for operation in self.request:
```